app.ts

```diff
@@ -0,0 +1,319 @@
+		});
+}
+
+function recordSelection() {
+	let recordNav: HTMLElement | null = document.getElementById("record-navigation-container"); // Navigation area
+	if (!recordNav) {
+		alert("The navigation is not working correctly refresh the page");
+	} else {
+		let singleRecordSelection = `
+		<button id="return-btn">Return</button>
+		<div id="user-input-data">
+			<div class="navigation-input-area-id" id="id">
+				<label class="record-labels" for="record-id">Enter record ID :</label>
+				<input type="text" name="record-id" id="record-id" class="navigation-input" value="0" />
+			</div>
+			<p class="amount-of-records"></p>
+		</div>
+		<button id="get-record-btn">See Record</button>`;
+
+		recordNav.innerHTML = singleRecordSelection;
+
+		let returnBtn: HTMLElement | null = document.getElementById("return-btn");
+		let recordIdInput: HTMLElement | null = document.getElementById("record-id");
+		let numberOfRows = Math.floor(window.innerHeight / 50);
+		let getSingleRecord: HTMLElement | null = document.getElementById("get-record-btn");
+
+		if (returnBtn !== null) {
+			// Resets to the first page
+			returnBtn.addEventListener("click", () => {
+				createNavigation();
+				fromNumber = 0;
+				getRecords(fromNumber, fromNumber + numberOfRows);
+			});
+		}
+
+		if (getSingleRecord !== null) {
+			getSingleRecord.addEventListener("click", () => {
+				let recordIdValue = (recordIdInput as HTMLInputElement).value;
+				fromNumber = Number(recordIdValue);
+
+				let toNumber = fromNumber + numberOfRows;
+				let finalRecord = recordNumberTotal - 1;
+
+				if (toNumber > finalRecord) {
+					toNumber = finalRecord;
+					fromNumber = toNumber - numberOfRows;
+				}
+
+				let check = ["undefined", "string", ""];
+
+				if (check.includes(typeof fromNumber) || fromNumber < 0) {
+					alert("Does not exists");
+					recordIdValue = "0";
+				} else if (typeof fromNumber === "number" && fromNumber >= 0) {
+					getRecords(fromNumber, toNumber);
+				}
+			});
+		}
+	}
+}
+
+function createHeadingGrid(headings: string) {
+	let headingColumns: HTMLElement | null = document.getElementById("column-headings-container"); // Headings
+	let headingsData = `<h1 class="column-heading">${headings}</h1>`;
+
+	if (headingColumns !== null) {
+		(headingColumns as HTMLDivElement).innerHTML += headingsData;
+	}
+}
+
+function recordCount(): Promise<void> {
+	return fetch("http://localhost:2050/recordCount", {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(checkResponseError)
+		.then((response: Response) => response.json())
+		.then((data: number) => {
+			recordNumberTotal = data;
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+function headingRowCreation(): Promise<void> {
+	return fetch("http://localhost:2050/columns", {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(checkResponseError)
+		.then((response: Response) => response.json())
+		.then((data: string) => {
```

Comment:
Not sure how `data` is a `string` when you have a `for-loop` over it on the next line...